pkg/operator/k8sutil/deployment.go

```diff
@@ -118,6 +118,15 @@ func UpdateDeploymentAndWait(context *clusterd.Context, modifiedDeployment *apps
 
 				return d, nil
 			}
+
+			// If ProgressDeadlineExceeded is reached let's fail earlier
+			// This can happen if one of the deployment cannot be scheduled on a node and stays in "pending" state
+			for _, condition := range d.Status.Conditions {
```

Comment:
Instead of checking for the progress deadline exceeded, could we remove the multiplier by 2 above on line 102? This would effectively reduce the retries to 10m instead of 20m, right? Not sure why we were multiplying by 2.

Reply:
Per discussion, understood that the 2x multiplier is to account for both the stop/start, and this condition check helps us only wait until the new pod timeout is hit.